### DISCUSSION TITLE
Single pane view of players on Jamulus

### DISCUSSION THREAD

**Discussion by rdica** [jamulussoftware/jamulus#3192] [General]
### Hi all!
There's a new site that provides a **single pane view** of players on Jamulus from all genres.

https://jamulus.diix.org

#### What It Does

- Displays occupied servers and their clients in all publicly available genres on a single page.
- Known bots and streamer clients highlighted for informational purposes. (requested)
- Data updated once a minute.
- Mobile site available. (requested)
- IPv6 capable.

#### What It Doesn't

- The site does not and will not add arbitrary judgements of servers, or filter them out, **for any reason**.
- Will not put bots/streamers on servers. There is nothing to firewall, or any changes required of server operators.

All information displayed is publicly available from directory servers, explorer.jamulus.io, or the Jamulus client itself.

We're not just site operators, we also run Jamulus servers, and play on Jamulus regularly.
We created the site so it'd be easier to see who was online without having to switch genres in the client, or on explorer.jamulus.io.

We do have future features planned. However, recording/streaming audio and chat from sessions are not among them. Other solutions already exist for that and we will never impose ourselves on operators, servers, or clients on them.

The site makes no claim of being a source of truth for who's online, only your Jamulus client can provide that, but we're confident it's as close as you can get without starting up a client 😃

Thanks to @softins for his work on [jamulus-php](https://github.com/softins/jamulus-php) without which we couldn't have created this site!

Hope you find it useful!

**Comment by pljones** [comment]
Does it have a name?

And it doesn't look like it's just the output of the PHP, so you should claim ownership of your work at the same point you credit @softins - and also put some contact details, too - even if it's just a reference to this discussion - so people know where to go with any questions or concerns.

(Ideally it'd have its own github or wherever repo for support.)

**Reply by pljones** [reply]
![image](https://github.com/jamulussoftware/jamulus/assets/1549463/29fe0969-efda-43d3-9d0e-96c4061b6c8a)
OK, so the **Known Bots `#cf0000`** and **Streamers `#a25c00`** are highlighted there but there's also this differentiated entry:
![image](https://github.com/jamulussoftware/jamulus/assets/1549463/b9787a3c-b0a2-4fe6-b2bd-b872597388de)
Is there any reason it stands out?

**Comment by rdica** [comment]
Thanks for your input! I will consider the points you made.

> Is there any reason it stands out?

No, it was a leftover of bot activity evaluation and was forgotten to be removed, thanks for pointing it out!

**Comment by mcfnord** [comment]
I've [guided users who want to locate bots](https://jamulus.live/) to your website in the _Ear and Lobby bots_ tab.

Maybe you could describe bots with labels. **stream, record, chat, & float** come to mind.

**Comment by pljones** [comment]
Got another anomoly:
**Who's On Jamulus**
![image](https://github.com/jamulussoftware/jamulus/assets/1549463/5dba1b82-15b0-4e03-9e22-362c21ab98ed)
vs
**Jamulus Explorer** - _AWS_
![image](https://github.com/jamulussoftware/jamulus/assets/1549463/93273bc2-45db-46d2-9b7a-7b2cea4c0827)

But I do see that
**Jamulus Explorer** - _Linode_ says
![image](https://github.com/jamulussoftware/jamulus/assets/1549463/34b2d099-147a-4211-a81a-caca64c28f4f)

so I guess you're using the (unreliable) Linode feed?
("NOTE: the London Linode server is currently having issues connecting to some directories.")

**Reply by mcfnord** [reply]
I haven't seen this "invisible list of connections" failure, but now consider the Linode servers.php instance "about as reliable" as the others. Tony and I added two more instances when issues emerged on Linode, but I've since noticed all instances seem to fail when one is failing. I introduced a fail-over system, but it had limited value since the other instances experienced similar problems. The modest caching in Tony's code might explain all the minor result differences. I wonder if Jamulus server network activity causes the data failures. It's puzzling because failures typically affect multiple directories, not just one, suggesting some pattern across multiple directories might be the cause.

**Reply by ann0see** [reply]
And you're sure it's not a routing issue?

**Reply by mcfnord** [reply]
It probably is. And on Tony's original servers.php instance, an interruption could last seconds or minutes or hours or days. I had forgotten about this until it happened again a few days ago. On the new instances, an interruption lasts seconds or minutes, never hours/days. So I guess there are differences. One twist: @rdica's website didn't appear to experience any interruption when mine clearly did, yet I'm reasonably sure we were both using https://jamulus.softins.co.uk/. He might have a smarter refresh strategy. I just slam the instance with 7 requests at once, and the default nginx chokes a little and complains in the log that beyond 5 you wait a bit. But that shouldn't impact routing to the directory server. I don't like being at the mercy of this unpredictable bug. I've made up mitigations but consider it a big risk.

**Comment by rdica** [comment]
I have moved the site to https://jamulusjams.com

If you're linking to me please update your links. I have an informational, temporary redirect in place which I expect to remove in the future.

Thanks

**Reply by seant-ca** [reply]
This is a great idea! I am new to Jamulus and I was going to request this feature but I was happy to see that someone has already thought of it! I was wondering if there is a way to list all connected clients, that may not already be connected to a server? I'm thinking that a general "lobby" or lounge would be useful where you can see who might be available to connect but is not yet in a server. Also, what are your thoughts on a general "community" or social-type platform for the main purpose of arranging jams with people who are online looking to connect. Do you think most people are connecting in private rooms? It would be nice to have a community or "directory" of sorts. Am I able to message you directly?

**Reply by softins** [reply]
Hi Sean, potential clients who are not connected to a server are not visible anywhere. It is only connecting to a server that makes them visible to others.

If you are on Facebook, there are a few groups where you might find people to connect with:
- "Jamulus (official group)", although that group's focus is more towards helping people with using Jamulus.
- "Jamulus WorldJam", which is fairly quiet since WorldJam finished a year ago, but there are a lot of people who lurk and might like to connect.
- "Jamulus Events & Connections", which is a general group for people to find others to play with, but again is fairly quiet.

**Reply by seant-ca** [reply]
Thanks for this info softins. It seems to be that the general challenge is that it's "hard" to collaborate live, especially if you need to have a certain amount of latency to be productive. Do you think that software like Jamulus became popular in the pandemic but now they are less used?

**Reply by rdica** [reply]
@seant-ca You can email me via the contact address on my site. I'm glad you find the site useful.

The quality of your experience on jamulus will depend on your behaviour when connected to a server. If you just start playing on top of everybody, you're likely to generate a negative response. Like any jam session, live or online, there's an etiquette you should follow. If you've been in bands/groups before, you're likely already aware of a lot of this.

When you connect to a server, check the chat window. Most servers have a welcome message that may outline expected behaviour, or provide links to useful resources like charts.

Listen first.

It's polite to ask to join, this shows you're respectful of the people already there.

If there are multiple players already playing the instrument you play, try to find a space musically where your instrument may fit. Work with the other players to determine which parts you could play if possible. Alternately, mute yourself and play along if too many instrumentalists/singers.

Don't play your instrument between songs, others are trying to talk and it's rude and disrespectful to play over them. If you must play between songs, mute yourself.

Players are people, not resources. Interact with them, get to know them, allow them to get to know you. This is crucial to joining/building a player community. A LOT of current players have been using Jamulus since at least the start of the pandemic. Over time they got to know each other and formed "groups" that play together. Over time you'll begin to recognize who plays with who, and if you join and listen, what styles of music they play.
From this interactivity you can define contact means (email/txt/signal/whatsapp/facebook/etc) and add a social media aspect to the jam activity. Some also use jitsi/zoom/etc (with mics muted of course) so you can see each other while you're playing.

Latency is an issue, if yours is high and you're trying to play with others, this can cause distractions and can generate a negative response. It's always a good idea to ask first if you're experiencing high latency, some are tolerant, some are not.

Above all, be respectful of everyone. Treat others as you wish to be treated. Simple respect goes a long way!

Cheers, I hope you have a great Jamulus experience!
